common/tests/dbconfig_test.cpp

```diff
@@ -0,0 +1,36 @@
+#include "common/dbconfig.h"
+
+#include <string>
+#include "gtest/gtest.h"
+
+namespace common {
+  /**
+   *  use https://www.freeformatter.com/json-escape.html#ad-output
+   *  to escape & unescape
+   */
+
+TEST(DBConfigTest, Basic) {
+  auto cfgMgr = DBConfigManager::get();
+  std::string content = R"({ 
+   "dataset": {
+    "testdataset" : {
+        "ack_mode" : 1
+        }
+    } 
+  })";
+  cfgMgr->loadJsonStr(content);
+  
+  EXPECT_EQ(cfgMgr->getReplicationMode("dbtest00001"), 0);
+  EXPECT_EQ(cfgMgr->getReplicationMode("testdataset00001"), 1);
+  EXPECT_NE(cfgMgr->getReplicationMode("db1_00001"), 1);
+  EXPECT_NE(cfgMgr->getReplicationMode("testdataset"), 1);
+
+  EXPECT_EQ(cfgMgr->getReplicationMode("dbtest00001", 1), 1);
+}
```

Comment:
a couple edge test cases:
- empty content
- no `dataset` filed in json

Reply:
done